waspc/cli/src/Wasp/Cli/Command/Call.hs

```diff
@@ -1,25 +1,56 @@
 module Wasp.Cli.Command.Call where
 
 data Call
-  = New Arguments
-  | Start
-  | StartDb
+  = New !NewArgs
+  | Start !StartArg
   | Clean
   | Uninstall
   | Compile
-  | Db Arguments -- db args
+  | Db !DbArgs
   | Build
   | Version
   | Telemetry
   | Deps
   | Dockerfile
   | Info
-  | PrintBashCompletionInstruction
-  | GenerateBashCompletionScript
-  | BashCompletionListCommands
-  | WaspLS
-  | Deploy Arguments -- deploy cmd passthrough args
-  | Test Arguments -- "client" | "server", then test cmd passthrough args
-  | Unknown Arguments -- all args
+  | Completion !CompletionArgs
+  | WaspLS !WaspLSArgs
+  | Deploy !Arguments -- deploy cmd passthrough args
+  | Test !TestArgs -- "client" | "server", then test cmd passthrough args
+  deriving (Show, Eq)
+
+data Shell = Bash | Zsh | Fish deriving (Show, Eq)
```

Comment:
Hmm, if we do this, I think we should move `CompletionArgs` closer to Completion stuff too. Maybe, it's better if we move all top-level arguments type closer to their parser or command runner (into their own type module) for consistency? :thinking:

Reply:
Hmm, if we do this, I think we should move `CompletionArgs` closer to Completion stuff too. Maybe, it's better if we move all top-level arguments type closer to their parser or command runner (into their own type module) for consistency. :thinking:

Reply:
Alright. I've tried with moving the `Shell` data type into `Wasp.Cli.Command.ShellCompletion.Parser`, however this would introduce cyclic dependency with `Wasp.Cli.Command.Call` because `CompletionArgs` in the module is depending on `Shell` as well. This also applies if I move `Shell` into `Wasp.Cli.Command.ShellCompletion`. Both of these parser and logic runner modules are depending on `CommandCall`.

So I think what I perhaps could do:

1. Leave `Shell` in `Wasp.Cli.Command.Call` as it is.
-or-
2. Introduce a new module `Wasp.Cli.Command.ShellCompletion.CompletionArgs` and bring both `Shell` and `CompletionArgs` into it.
    - Make `Wasp.Cli.Command.Call` and Completion modules import them normally.
    - I've tried 2. and it works all right (with ParserTest too), but I find it odd to see only CompletionArgs in their own specific module. So, I wonder if we should move the rest of command args type to their own module as well, following the same approach.

Reply:
Ha, this is getting a bit complex!
And this really ties in with the other big comment I left, about deciding where these data structures go.

We gave it a try, but since it is not clear, maybe it is best to just leave it as it is, until with time it becomes clearer how we can handle it. What do you think?

Reply:
Sure, we can jump directly to the big comment and solve this one over there.

Reply:
We can maybe even also give up on the big comment, that is what I was trying to suggest :D.

Reply:
Ohh! Okay then sure. :)